odoo/addons/base/models/ir_fields.py

```diff
@@ -604,7 +604,7 @@ def _str_to_one2many(self, model, field, records):
         def log(f, exception):
             if not isinstance(exception, Warning):
                 current_field_name = self.env[field.comodel_name]._fields[f].string
-                arg0 = exception.args[0] % {'field': '%(field)s/' + current_field_name}
+                arg0 = exception.args[0].replace('%(field)s', '%(field)s/' + current_field_name)
```

Comment:
Damned, I had to read the original code 5 or 6 times before understanding its intention.

Reply:
Yeah, using recursive calls with escaping and successive formatting from values coming from various places is not the best design.
I doubt this is the last bug in this code (if we don't take into account that this fix is leaving a bug since it still isn't escaped properly)

Reply:
It wouldn't work for `self.import_(['value/m2o'], [['%(field)s']])` (literal `'%(field)s'`) but going from "impossible to use any %" to "impossible to use %(field)s" is better I guess.